components/aws/sagemaker/common/_utils.py

```diff
@@ -113,17 +122,49 @@ def print_logs_for_job(cw_client, log_grp, job_name):
         logging.error(e)
 
 
-def get_sagemaker_client(region, endpoint_url=None):
+def get_boto3_session(region, role_arn=None):
+    """Creates a boto3 session, optionally assuming a role"""
+
+    # By default return a basic session
+    if role_arn is None:
+        return Session()
+
+    def refresh():
+        client = boto3.client("sts", region_name=region,
+                                endpoint_url='https://sts.' + region + '.amazonaws.com')
+        response = client.assume_role(RoleArn=role_arn, RoleSessionName=ASSUME_SESSION_NAME_FORMAT).get("Credentials")
+        return dict(
+            access_key=response.get("AccessKeyId"),
+            secret_key=response.get("SecretAccessKey"),
+            token=response.get("SessionToken"),
+            expiry_time=response.get("Expiration").isoformat()
+        )
+
+    session_credentials = RefreshableCredentials.create_from_metadata(
```

Comment:
Nice!

Reply:
I referenced a number of documents about this `RefreshableCredentials` class. The main one was [this one](https://dev.to/li_chastina/auto-refresh-aws-tokens-using-iam-role-and-boto3-2cjf) and [the actual implementation](https://github.com/boto/botocore/blob/develop/botocore/credentials.py#L353). It has constructor arguments for setting refresh frequencies but I left them alone.